fastjet/internal/delaunay/delaunay_test.go

```diff
@@ -0,0 +1,289 @@
+			x := tx*c - ty*s
+			y := tx*s + ty*c
+			points = append(points, NewPoint(x, y))
+		}
+	}
+	return points
+}
+
+func TestHierarchicalDelaunayGrid(t *testing.T) {
+	const degrees = math.Pi / 180
+	const n = 10
+	ps := grid(n, n, 10*degrees)
+	d := HierarchicalDelaunay()
+	for _, p := range ps {
+		d.Insert(p)
+	}
+}
+
+func benchmarkHierarchicalDelaunayInsertion(i int, b *testing.B) {
+	ps := make([]*Point, i)
+	for j := 0; j < i; j++ {
+		x := rand.Float64() * 1000
+		y := rand.Float64() * 1000
+		ps[j] = NewPoint(x, y)
+	}
+	b.ReportAllocs()
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		d := HierarchicalDelaunay()
+		for _, p := range ps {
+			d.Insert(p)
+		}
+	}
+}
+
+func BenchmarkHierarchicalDelaunayInsertion5(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(5, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion6(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(6, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion7(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(7, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion8(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(8, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion9(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(9, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion10(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(10, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion11(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(11, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion12(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(12, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion13(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(13, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion14(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(14, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion15(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(15, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion16(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(16, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion17(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(17, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion18(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(18, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion19(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(19, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion20(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(20, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion21(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(21, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion22(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(22, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion23(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(23, b)
+}
+
+func BenchmarkHierarchicalDelaunayInsertion24(b *testing.B) {
+	benchmarkHierarchicalDelaunayInsertion(24, b)
```

Comment:
I mean that the benchmark should be done on some representative problem sizes. This suggests that the problem sizes will be very small.